tests/apiperf.cpp

```diff
@@ -0,0 +1,249 @@
+                ASSERT_NEQ(mh, null_mh);
+
+                void *map_d_ptr  = NULL;
+                ASSERT_EQ(gdr_map(g, mh, &map_d_ptr, actual_pin_size), 0);
+
+                gdr_info_t info;
+                ASSERT_EQ(gdr_get_info(g, mh, &info), 0);
+                ASSERT_EQ(gdr_unmap(g, mh, map_d_ptr, actual_pin_size), 0);
+                ASSERT_EQ(gdr_unpin_buffer(g, mh), 0);
+            }
+
+            for (iter = 0; iter < num_iters; ++iter) {
+
+                clock_gettime(MYCLOCK, &beg);
+                BREAK_IF_NEQ(gdr_pin_buffer(g, d_A, actual_pin_size, 0, 0, &mh), 0);
+                clock_gettime(MYCLOCK, &end);
+                delta_lat_us = ((end.tv_nsec-beg.tv_nsec)/1000.0 + (end.tv_sec-beg.tv_sec)*1000000.0);
+                pin_lat_us += delta_lat_us;
+                ASSERT_NEQ(mh, null_mh);
+                lat_arr[iter] = delta_lat_us;
+                min_lat = delta_lat_us < min_lat ? delta_lat_us : min_lat;
+                max_lat = delta_lat_us > max_lat ? delta_lat_us : max_lat;
+
+                void *map_d_ptr  = NULL;
+                clock_gettime(MYCLOCK, &beg);
+                ASSERT_EQ(gdr_map(g, mh, &map_d_ptr, actual_pin_size), 0);
+                clock_gettime(MYCLOCK, &end);
+                delta_lat_us = ((end.tv_nsec-beg.tv_nsec)/1000.0 + (end.tv_sec-beg.tv_sec)*1000000.0);
+                map_lat_us += delta_lat_us;
+
+                gdr_info_t info;
+                clock_gettime(MYCLOCK, &beg);
+                ASSERT_EQ(gdr_get_info(g, mh, &info), 0);
+                clock_gettime(MYCLOCK, &end);
+                delta_lat_us = ((end.tv_nsec-beg.tv_nsec)/1000.0 + (end.tv_sec-beg.tv_sec)*1000000.0);
+                inf_lat_us += delta_lat_us;
+
+                clock_gettime(MYCLOCK, &beg);
+                ASSERT_EQ(gdr_unmap(g, mh, map_d_ptr, actual_pin_size), 0);
+                clock_gettime(MYCLOCK, &end);
+                delta_lat_us = ((end.tv_nsec-beg.tv_nsec)/1000.0 + (end.tv_sec-beg.tv_sec)*1000000.0);
+                unmap_lat_us += delta_lat_us;
+
+                clock_gettime(MYCLOCK, &beg);
+                ASSERT_EQ(gdr_unpin_buffer(g, mh), 0);
+                clock_gettime(MYCLOCK, &end);
+                delta_lat_us = ((end.tv_nsec-beg.tv_nsec)/1000.0 + (end.tv_sec-beg.tv_sec)*1000000.0);
+                unpin_lat_us += delta_lat_us;
+            }
+
+            pin_lat_us /= iter;
+            map_lat_us /= iter;
+            inf_lat_us /= iter;
+            unpin_lat_us /= iter;
+            unmap_lat_us /= iter;
+
+            printf("Size(B)\tpin.Time(us)\tmap.Time(us)\tget_info.Time(us)\tunmap.Time(us)\tunpin.Time(us)\n");
+            printf("%zu\t%f\t%f\t%f\t%f\t%f\n",
+                    actual_pin_size, pin_lat_us, map_lat_us, inf_lat_us, unmap_lat_us, unpin_lat_us);
+            pin_request_size <<= 1;
+
+            print_histogram(lat_arr, num_iters, bin_arr, num_bins, min_lat, max_lat);
```

Comment:
Before printing out the histogram, can you also print a header "Histogram of ..."?